src/nunit.analyzers/Helpers/AssertExpressionHelper.cs

```diff
@@ -92,5 +119,105 @@ public static IEnumerable<ExpressionSyntax> SplitConstraintByOperators(Expressio
                 yield return constraintExpression;
             }
         }
+
+        /// <summary>
+        /// If constraint expression is combined using And, Or, With properties - 
+        /// returns parts of expression split by those properties.
+        /// /// </summary>
+        private static IEnumerable<ExpressionSyntax> SplitConstraintByConstraintExpressionOperators(ExpressionSyntax constraintExpression)
+        {
+            // e.g. Does.Contain(a).IgnoreCase.And.Contain(b).And.Not.Null
+            // --> 
+            // Does.Contain(a).IgnoreCase,
+            // Does.Contain(a).IgnoreCase.And.Contain(b)
+            // Does.Contain(a).IgnoreCase.And.Contain(b).And.Not.Null
+
+            // (You cannot separate only Not.Null part in any way)
+
+            var callChainParts = constraintExpression.SplitCallChain();
+
+            for (var i = 1; i < callChainParts.Count; i++)
+            {
+                if (IsConstraintExpressionOperator(callChainParts[i]))
+                {
+                    yield return callChainParts[i - 1];
+                }
+            }
+
+            yield return constraintExpression;
+        }
+
+        private static (ExpressionSyntax expectedArgument, IMethodSymbol constraintMethod)? GetExpectedArgumentFromConstraintPart(
+            ExpressionSyntax constraintPart,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            // Each constraint part might have only one method, accepting 'expected' argument.
```

Comment:
I'm unsure about the might in the statement.

Reply:
e.g. Is.Not.Null` has none

Reply:
Instead of "Each constraint part might have only one method, accepting 'expected' argument." could we write "Each constraint part will at most have one method accepting an 'expected' argument."